Add formatting check to Travis build (#852)
tjcsl/ion: docs/developing/styleguide.rst

        flake8 --max-line-length 150 --exclude=*/migrations/* .
        pylint --jobs=0 --disable=fixme,broad-except,global-statement,attribute-defined-outside-init intranet/
        isort --check --recursive intranet
        ./scripts.format.sh

> Note: The ``./scripts/format.sh`` check is run in such a way that the build will fail if it has to makea any changes.

Reviewer: ```suggestion
Note: When the ``./scripts/format.sh`` check is run, the build will fail if it has to make any changes.
```
Also add a blank line above this line

``flake8`` is a PEP8 style checker, ``pylint`` is a linter (but it also enforces some PEP8 conventions), and ``isort``, when called with these options, checks that all imports are sorted alphabetically.

``./scripts/format.sh`` runs ``black intranet && autopep8 --in-place --recursive intranet && isort --recursive intranet``. The reason for the multiple commands is that ``black`` introduces certain formatting changes which ``flake8``/``pylint`` do not agree with (and offers no options to change them), so we have ``autopep8`` fix it.

It is recommended that you run all of these locally before opening a pull request (though the Ion developers sometimes skip running the ``pylint`` check locally because it takes a long time to run). All of them are intended to be run from the root directory of the Git repository.

> If ``flake8`` or ``pylint`` throw errors, the error messages are usually human-readable. if ``isort`` gives any errors, you can have it automatically correct the order of all imports by running ``isort --recursive intranet``. If the running of ``scripts/format.sh`` results in any changes, you can simply run  ``./scripts/format.sh`` to fix your formatting.

Reviewer: `If running ``scripts.format.sh`` results in any changes`...
